src/biogeochem/test/DustEmis_test/test_DustEmisZender2003.pf

```diff
@@ -0,0 +1,483 @@
+module test_DustEmisZender2003
+
+  ! Tests of DustEmisZender2003
+
+  use funit
+  use unittestSubgridMod
+  use unittestSimpleSubgridSetupsMod, only : setup_single_veg_patch
+  use clm_varpar, only : nlevsoi, nlevgrnd, nlevsno, clm_varpar_init, ndst
+  use clm_varctl, only : soil_layerstruct_predefined, create_crop_landunit, use_crop, create_crop_landunit
+  use clm_varcon, only : clm_varcon_init, clm_varcon_clean
+  use SnowHydrologyMod, only : InitSnowLayers, SnowHydrologySetControlForTesting, SnowHydrologyClean
+  use DustEmisBase
+  use DustEmisZender2003
+  use shr_kind_mod , only : r8 => shr_kind_r8
+  use unittestFilterBuilderMod, only : filter_from_range
+  use atm2lndType, only : atm2lnd_type, atm2lnd_params_type
+  use SoilStateType, only : soilstate_type
+  use CanopyStateType, only : canopystate_type
+  use TemperatureType, only : temperature_type
+  use WaterType, only : water_type
+  use FrictionVelocityMod, only : frictionvel_type
+  use unittestWaterTypeFactory, only : unittest_water_type_factory_type
+  use SoilStateInitTimeConstMod, only : ThresholdSoilMoistZender2003, ThresholdSoilMoistKok2014, MassFracClay
+  use DustEmisFactory, only : create_dust_emissions
+
+  implicit none
+
+  real(r8), parameter :: tol = 1.e-18_r8
+
+  @TestCase
+  type, extends(TestCase) :: TestDustEmisZender2003
+     class(dust_emis_base_type), allocatable :: dust_emis
+     integer, allocatable :: filter_nolakep(:)      ! non-lake filter (patches)
+     integer :: num_nolakep              ! number of patches in non-lake filter
+     type(atm2lnd_type) :: atm2lnd_inst
+     type(soilstate_type) :: soilstate_inst
+     type(canopystate_type) :: canopystate_inst
+     type(temperature_type) :: temperature_inst
+     type(unittest_water_type_factory_type) :: water_factory
+     type(water_type) :: water_inst
+     type(frictionvel_type) :: frictionvel_inst
+   contains
+     procedure :: setUp
+     procedure :: tearDown
+     procedure :: setupEnvironment
```

Comment:
Make an issue that these setup subroutines should be put into some type of unit-test shared place so that both the Zender and Leung methods can use them.

Reply:
I actually just went ahead and did this and put the shared part into the unit_test_shr directory.